pkg/test/framework/components/echo/kube/deployment_test.go

```diff
@@ -0,0 +1,55 @@
+package kube
+
+import (
+	"testing"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/core/image"
+)
+
+var (
+	settings *image.Settings = &image.Settings{
+		Hub:        "testing.hub",
+		Tag:        "latest",
+		PullPolicy: "Always",
+	}
+)
+
+func TestDeploymentYAML(t *testing.T) {
```

Comment:
It would also be good to see an actual integration test switched over to use this. Perhaps traffic shifting?

Reply:
I originally find this need when I'm writing a test. And seems like implementing this PR is non-trivial enough to become a separate PR, so i have this test framework implemented separately to make reviewer easier.

I'll see how big that becomes once I'm done with this TF feature.

Reply:
i'll defer the usage in a separate PR. this is big enough. I did test it with my usage if you are curious, https://github.com/istio/istio/pull/21369/commits/9a20acc0d17d7aab58cc58f917f0cdd593ac7712#diff-82429f8ee906a654d6936dd58b9fd67a

this is reverted in my pr.